File: Matrix.cpp
#include <iostream>
using namespace std;

int main()
{
    int n, m;
    cout << "Enter number of rows : ";
    cin >> n;
    cout << "Enter number of columns : ";
    cin >> m;
    int a = n;
    int b = m;
    int c = n;
    int d = m;
    int arr[n][m];
    int sort[a][b];
    int MM[c][d];

    // Matrices Input
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cout << "Enter element"
                 << "(" << i + 1 << ")"
                 << "(" << j + 1 << ")"
                 << ": ";
            cin >> arr[i][j];
            sort[i][j] = arr[i][j];
        }
    }
    cout << endl;

    // row-wise sorting
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            for (int k = 0; k < m - j - 1; k++)
            {
                if (sort[i][k] > sort[i][k + 1])
                {
                    swap(sort[i][k], sort[i][k + 1]);
                }
            }
        }
    }

    // Original Matrix Output
    cout << "Original Matrix >>> " << endl;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cout << arr[i][j] << " ";
        }
        cout << endl;
    }

    // Sorted Matrix Output
    cout << "Sorted Matrix >>>" << endl;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cout << sort[i][j] << " ";
        }
        cout << endl;
    }

    // Adduct Matrix Output
    cout << "Adduct Matrix >>>" << endl;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cout << sort[i][j] + arr[i][j] << " ";
        }
        cout << endl;
    }

    // Matrix Multiplication
    cout << "Multiplied Matrix >>> " << endl;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            int value;
            for (int k = 0; k < n; k++)
            {
                value = +arr[i][k] * sort[k][j];
            }
            MM[i][j] = value;
        }
    }
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cout << MM[i][j] << " ";
        }
        cout << endl;
    }
    return 0;
}

File: Matrix.java
import java.util.*;

public class Matrix {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of rows : ");
        int rows = scanner.nextInt();
        System.out.print("Enter the number of columns : ");
        int columns = scanner.nextInt();

        int[][] originalMatrix = new int[rows][columns];
        int[][] sortedMatrix = new int[rows][columns];
        int[][] adductMatrix = new int[rows][columns];
        int[][] multiMatrix = new int[rows][columns];
        int[] highValue = new int[rows];

        // Input the matrix
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                System.out.print(String.format("Enter element %d,%d : ", i + 1, j + 1));
                originalMatrix[i][j] = scanner.nextInt();
                sortedMatrix[i][j] = originalMatrix[i][j]; // Copy for sorting
            }
        }

        // Sort each row of the sorted matrix using bubble sort
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns - 1; j++) {
                for (int k = 0; k < columns - j - 1; k++) {
                    if (sortedMatrix[i][k] > sortedMatrix[i][k + 1]) {
                        int temp = sortedMatrix[i][k];
                        sortedMatrix[i][k] = sortedMatrix[i][k + 1];
                        sortedMatrix[i][k + 1] = temp;
                    }
                }
            }
        }

        // Print the original matrix
        System.out.println("The Original Matrix is : ");
        printMatrix(originalMatrix);

        // Print the sorted matrix
        System.out.println("The Sorted Matrix is : ");
        printMatrix(sortedMatrix);

        // Add the original and sorted matrices
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                adductMatrix[i][j] = originalMatrix[i][j] + sortedMatrix[i][j];
            }
        }

        // Print the adduct matrix
        System.out.println("The Adduct Matrix is : ");
        printMatrix(adductMatrix);

        // Multiply the original and sorted matrices
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                for (int k = 0; k < rows; k++) {
                    multiMatrix[i][j] += originalMatrix[i][k] * sortedMatrix[k][j];
                }
            }
        }
        System.out.println("The Multiplied Matrix is : ");
        printMatrix(multiMatrix);

        // Find the row with the highest value
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                highValue[i] += adductMatrix[i][j];
            }
        }

        int rowIndexWithMaxValue = 0;
        int maxValue = highValue[0];
        for (int i = 1; i < rows; i++) {
            if (highValue[i] > maxValue) {
                maxValue = highValue[i];
                rowIndexWithMaxValue = i;
            }
        }
        System.out.println();
        System.out.println(String.format("Row with the highest value is : %d", rowIndexWithMaxValue + 1));
        System.out.println();

        scanner.close();
    }

    private static void printMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            for (int element : row) {
                System.out.print(element + " ");
            }
            System.out.println();
        }

    }
}
